pkgs/development/cuda-modules/packages/autoAddCudaCompatRunpath/package.nix

```diff
@@ -13,7 +13,7 @@ makeSetupHook {
   propagatedBuildInputs = [ autoFixElfFiles ];
 
   substitutions = {
-    libcudaPath = "${cuda_compat}/compat";
+    libcudaPath = if cuda_compat != null then "${cuda_compat}/compat" else null;
```

Comment:
Hm I thought this should've been sufficient: https://github.com/NixOS/nixpkgs/pull/405031/files#diff-f0756e113cff4d541c62e50496aca47492801463cb75266b75732935906ffe63R19-R27

Reply:
Tldr: we've had this exact patch before, I'm hoping to eliminate this hacks by setting the platforms right